### ISSUE TITLE
Per-domain configuration settings not working

### ISSUE THREAD

**Issue by mw-a** [himmelblau-idm/himmelblau#333]
**Description**
The example configuration suggests that settings such as `home_attr`, `shell` and colleagues can be customised per domain:

https://github.com/himmelblau-idm/himmelblau/blob/3b19d0b6cca6e40b1206fc855c767fe04e4b3907/src/config/himmelblau.conf.example#L112

In reality this does not work in my testing.

Looking at the code it appears that it cannot currently work because the respective functions are only called once when instantiating `Resolver` and without domain:

https://github.com/himmelblau-idm/himmelblau/blob/3b19d0b6cca6e40b1206fc855c767fe04e4b3907/src/daemon/src/daemon.rs#L1020

Is this assessment correct?

It's no real issue for my current testing but would appear to be a bit of a documentation bug still.

**Environment**
- **Linux Distro**: RockyLinux 8
- **Package source (distro package/github release/self built)**: self built
- **Himmelblau Version**: 0.8.0 as of 3b19d0b6cca6e40b1206fc855c767fe04e4b3907

**Comment by dmulder** [comment]
Yes, I would call it a documentation bug at this point. In reality, those settings are **per-tenant**, so those settings are misleading.

**Comment by dmulder** [comment]
I think I could fix it so the settings _do_ apply per-domain name. Do you think that would be preferred?

**Comment by dmulder** [comment]
> Looking at the code it appears that it cannot currently work because the respective functions are only called once when instantiating `Resolver` and without domain:
> 
> https://github.com/himmelblau-idm/himmelblau/blob/3b19d0b6cca6e40b1206fc855c767fe04e4b3907/src/daemon/src/daemon.rs#L1020
> 

FYI, that's actually just the defaults being loaded. The user shell (and other user attributes) are set here:

https://github.com/himmelblau-idm/himmelblau/blob/3b19d0b6cca6e40b1206fc855c767fe04e4b3907/src/common/src/idprovider/himmelblau.rs#L1532

**Comment by dmulder** [comment]
> _Actually_, now I think of it, for CN in particular I would want a behaviour similar to `cn_name_mapping` where it's only used for the primary domain to avoid collisions. But then if I have machines with different primary domains, I would need the settings reversed if the home directories are on a shared network filesystem to get a consistent mapping. That might be an argument for per-domain settings. But I have not thought this through and don't know the behaviour of CN in `home_attr` in relation to primary/secondary domains in the first place.

IIRC, `cn_name_mapping` is only working for the primary domain, and not the additional custom domains in the same tenant. FYI, technically the 'primary' domain can be set to any of the domains for the tenant. It doesn't matter which one you set as primary.
You make a good point though. There are instances where it makes sense to set the `home_attr` differently between a primary domain and the other custom domains.

**Comment by dmulder** [comment]
The per-domain settings have been removed.
